src/main/scala/medeia/generic/CoproductDerivationOptions.scala

```diff
@@ -0,0 +1,8 @@
+package medeia.generic
+
+case class CoproductDerivationOptions[A](
```

Comment:
I think we could improve the name here.  `Coproduct` leaks that we use shapeless, but that's a detail, we could also be using magnolia or sth else.

`SealedTraitDerivationOptions`? `AdtDerivationOptions`? `TaggedUnionDerivationOptions`?  really hard to find a good name :sweat_smile:

Reply:
That would be easier if only one Options class existed...

Reply:
SealedTrait it is until a decision